src/ai/backend/client/cli/image.py

```diff
@@ -92,3 +92,40 @@ def forget(reference_or_id, arch):
             print_done(f"Image forgotten: {reference_or_id}")
         else:
             print_fail("Image forget has failed: {0}".format(result["msg"]))
+
+
+@image.command()
+@click.argument("reference_or_id", type=str)
+@click.option("--arch", type=str, default=None, help="Set an explicit architecture.")
+@click.option(
+    "--remove-from-registry",
+    is_flag=True,
+    default=False,
+    help="If set, removes the image from the registry. Only supported in Harbor registry.",
+)
+def purge(reference_or_id, arch, remove_from_registry):
+    """Delete image from server. This command will only work for image customized by user
+    unless callee has superadmin privileges.
+
+    REFERENCE_OR_ID: Canonical string of image (<registry>/<project>/<name>:<tag>)"""
+    with Session() as session:
+        try:
+            image_id = get_image_id(session, reference_or_id, architecture=arch)
+        except BackendAPIError:
+            print_fail("Could not find image.")
+            if not arch:
+                print_warn(
+                    "`arch` parameter not passed. If you are trying to resolve image via its canonical string you have to specify which architecture are you trying to manipulate with."
+                )
```

Comment:
I feel like this would be an inappropriate place to warn when there is no arch, is there a reason to warn at this point?

Reply:
This should be refactored as following work.